src/HttpHeaders.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace P4\MasterTheme;
+
+/**
+ * Class HttpHeaders
+ */
+class HttpHeaders {
+	/**
+	 * Headers constructor.
+	 */
+	public function __construct() {
+		add_action( 'wp_headers', [ $this, 'send_content_security_policy_header' ], 10, 1 );
+	}
+
+
+	/**
+	 * Send Content Security Policy (CSP) HTTP headers.
+	 */
+	public function send_content_security_policy_header() {
+		$default_allowed_frame_ancestors = [ '\'self\'' ];
+
+		// Filter hook to allow adding trusted frame ancestors.
+		$additional_allowed_frame_ancestors = apply_filters( 'planet4_csp_allowed_frame_ancestors', [] );
+
+		$allowed_frame_ancestors = array_merge( $default_allowed_frame_ancestors, (array) $additional_allowed_frame_ancestors );
```

Comment:
Maybe best to not cast it to an array and just document the filter needs to return an array of strings (and maybe check if it actually is one before doing the merge). Casting to an array will do some unwanted conversions. For example if the filter by accident returns an object (like a `WP_ERROR`), then it will be converted to an array where the keys are the keys of the object (including private properties). See https://www.php.net/manual/en/language.types.array.php#language.types.array.casting

Reply:
Removed the cast and instead checked for `$additional_allowed_frame_ancestors` to be an array. 

Not sure what to do with the documentation. I used a one line comment for consistency with the rest of the master theme. Considering the limited space I didn't mention the return type, every filter function is expected to return the first argument.

I could change it to how the [Wordpress Handbook](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#4-hooks-actions-and-filters) suggests documenting a hook. It also only specifies the parameter and assumes the return value is the same. But then it's inconsistent with the rest of the code. What's better?

Example:
```
/**
 * Filter hook to allow adding trusted frame ancestors to the Content Security Policy.
 * 
 * @param array $additional_allowed_frame_ancestors Array of domains to whitelist as frame ancestors.
 */
```

Reply:
Well, we don't enforce how phpdoc should look for filters, so I guess we can indeed follow how WordPress does it.